app/controllers/posts_controller.rb

```diff
@@ -86,6 +86,14 @@ def destroy
     redirect_to root_path
   end
 
+  def post_title_search
+    @posts = Post.where("title ILIKE ?", "%#{params[:search]}%")
+    @communities= Community.where("name ILIKE ?", "%#{params[:search]}%")
+    @accounts=Account.where("username ILIKE ?", "%#{params[:search]}%")
+    @comments=Comment.where("message  ILIKE ?", "%#{params[:search]}%")
```

Comment:
There seem to be too many queries in here, we can definitely optimize that. We can load what is required for the first tab and then the other tabs can be loaded via ajax one by one.